File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        return searchRecur(nums, target, 0, nums.size() - 1);
    }
    
    int searchRecur(vector<int>& nums, int target, int low, int high) {
        if (low > high) {
            return -1;
        } else {
            auto mid = (low + high) / 2;
            
            if (nums[mid] == target) {
                return mid;
            }
            else if (target > nums[mid]) {
                return searchRecur(nums, target, mid + 1, high);
            }
            else {
                return searchRecur(nums, target, low, mid - 1);
            }
        }
    }
};

File: 0704-binary-search/0704-binary-search.ts
function search(nums: number[], target: number): number {
    
    function bs(lo: number, hi: number): number {
        if (lo > hi) {
            return -1;
        }
        
        const mid: number = Math.floor((lo + hi) / 2);
        if (target > nums[mid]) {
            return bs(mid + 1, hi);
        } else if (target < nums[mid]) {
            return bs(lo, mid - 1);
        } else {
            return mid;
        }
    }
    return bs(0, nums.length - 1);
};